core/src/main/kotlin/knote/data/NotebookImpl.kt

```diff
@@ -0,0 +1,32 @@
+package knote.data
+
+import knote.api.Notebook
+import knote.api.PageManager
+import knote.script.NotebookScript
+import knote.util.MutableKObservableObject
+import java.io.File
+import kotlin.script.experimental.api.ScriptDiagnostic
+
+class NotebookImpl(
+    override val id: String,
+    file: File,
+    fileContent: String = file.readText(),
+    compiledScript: NotebookScript? = null,
+    reports: List<ScriptDiagnostic>? = null,
+    pageManager: PageManager? = null
+) : Notebook {
+    override val fileObject: MutableKObservableObject<Notebook, File> = MutableKObservableObject(file)
+    override val fileContentObject: MutableKObservableObject<Notebook, String> = MutableKObservableObject(fileContent)
+    override val compiledScriptObject: MutableKObservableObject<Notebook, NotebookScript?> =
+        MutableKObservableObject(compiledScript)
```

Comment:
actually that does not work.. it needs to be declared as a MutableKObservable, because the interface just specifies a KObservable,
if i leave out the type here i have to cast this every time i want to mutate from within this object

Reply:
ahhhh okay. good catch! never mind on that